WebRoot/js/zimbraMail/share/view/dialog/ZmUploadDialog.js

```diff
@@ -479,28 +477,37 @@ ZmUploadDialog.prototype._addFileInputRow = function(oneInputOnly) {
     }
 };
 
-ZmUploadDialog.prototype._handleFileSize =
-function(inputEl, sizeEl){
+ZmUploadDialog.prototype._handleFileSelection =
+function(inputEl, fileDetailEl){
 
     var files = inputEl.files;
-    if(!files) return;
+    if(!files || !files.length){
+        fileDetailEl && (fileDetailEl.innerHTML = "");
+        return;
+    }
 
-    var sizeStr = [], className, totalSize =0;
+    var sizeStr = [], className, fileNameDetail, totalSize =0;
     for(var i=0; i<files.length;i++){
         var file = files[i];
         var size = file.size || file.fileSize /*Safari*/ || 0;
-	    var aCtxt = ZmAppCtxt.handleWindowOpener();
+        var aCtxt = ZmAppCtxt.handleWindowOpener();
         if(size > aCtxt.get(ZmSetting.DOCUMENT_SIZE_LIMIT))
             className = "RedC";
         totalSize += size;
     }
 
-    if(sizeEl) {
-        sizeEl.innerHTML = "  ("+AjxUtil.formatSize(totalSize, true)+")";
+    if(files.length > 1) {
```

Comment:
It's not about scope but caching the object property into a variable as files.length is used multiple times. It's a good practice to cache object property into a local variable if they are used multiple times.
Please refer :---

https://gamealchemist.wordpress.com/2016/04/15/writing-efficient-javascript-a-few-tips/

Inline content from this page :--

_Best example of such habit is to cache a repeated computation into a var : it will run faster, require less typing / less reading, and when hovering your mouse over the var in your debugger, you can check the cached value at once when debugging…

Rule is to cache everything you’re going to use again into a local var._